### PR TITLE
Color generator

### PR DESCRIPTION
**Pull request by RyanDBarnett**
# Description
Generate random colors and display them when color generator component mounts.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other, Please explain:
`Your message here`

## Has This Been Tested?

- [X] No
- [ ] Yes

# Checklist:

- [X] My code has no unused/commented out code
- [X] My code has no console logs
- [X] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by ehk9000** [comment]
I went ahead and resolved the merge requests with Justin. Looking good so far, would love to go over this tonight or tomorrow just to be brought up to speed on the logic

**Comment by ehk9000** [comment]
@RyanDBarnett

File: src/components/color_generator/index.js [hunk 1]
```diff
@@ -6,49 +6,76 @@ class ColorGenerator extends Component {
...
-      color4: {hex: '', lockType: 'lock-open'},
-      color5: {hex: '', lockType: 'lock-open'}
+      colors: [{}, {}, {}, {}, {}]
     }
   }
 
-  handleName = (e) => {
-    const { name, value } = e.target;
+  componentDidMount() {
+    this.generateColors()
```
Comment (ehk9000) [inline src/components/color_generator/index.js]: Do we want this to be invoked on componentDidMount even when the user is loading an existing palette?

File: src/components/color_generator/index.js [hunk 2]
```diff
@@ -6,49 +6,76 @@ class ColorGenerator extends Component {
...
+    const i = e.target.closest('.color-field').getAttribute('data-key');
+    const { colors } = this.state;
+    colors[i].locked = !colors[i].locked;
+    this.setState({ colors })
   }
 
   render() {
-    let {palette_name, color1, color2, color3, color4, color5} = this.state;
+    const {palette_name, colors} = this.state;
+    const colorFields = colors.map((colorObj, index) => {
```
Comment (ehk9000) [inline src/components/color_generator/index.js]: Could you explain what this map is doing next time we talk?

File: src/components/color_generator/index.js (hunk 1 above)
Comment (SiimonStark) [reply src/components/color_generator/index.js]: I didn't think of that, that's a good point.
- We could maybe refactor it to where: if Hex <string> is blank, then run generate colors.

File: src/components/color_generator/index.js [hunk 3]
```diff
@@ -6,49 +6,76 @@ class ColorGenerator extends Component {
...
   }
 
-  handleName = (e) => {
-    const { name, value } = e.target;
+  componentDidMount() {
+    this.generateColors()
+  }
 
-    this.setState({ [name]: value })
+  generateColors = () => {
```
Comment (SiimonStark) [inline src/components/color_generator/index.js]: Nice case of Single Responsibility, along with generateHex

File: src/components/color_generator/index.js [hunk 4]
```diff
@@ -6,49 +6,76 @@ class ColorGenerator extends Component {
...
 
-    let color = this.state[name];
-    color.hex = value;
+  handleName = (e) => {
+    const { name, value } = e.target;
+    this.setState({ [name]: value })
+  }
 
-    this.setState({ [name]: color });
+  updateColor = (e) => {
```
Comment (SiimonStark) [inline src/components/color_generator/index.js]: Could you please go over this function with me, when you get a chance?

File: src/components/color_generator/index.js (hunk 2 above)
Comment (SiimonStark) [reply src/components/color_generator/index.js]: @ehk9000 Essentially it's shortening up the code that I had written, allowing for each of the fieldsets to be dynamic.

File: src/components/color_generator/index.js [hunk 5]
```diff
@@ -60,101 +87,7 @@ class ColorGenerator extends Component {
           placeholder="Palette Name"
           maxLength="10"
         />
-        <fieldset className="color-field">
```
Comment (SiimonStark) [inline src/components/color_generator/index.js]: HUGE delete, thank you for making my code dynamic. I had thought about it after the fact last night, but my primary goal was to get things usable and on the page.

**Pull request merged**